sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1534,6 +1534,9 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
       // If the projection list contains Stars, expand it.
       case p: Project if containsStar(p.projectList) =>
         p.copy(projectList = buildExpandedProjectList(p.projectList, p.child))
+      // If the filter list contains Stars, expand it.
```

Comment:
shall we apply it for all operators?

Reply:
Define all. I don't think it's useful fro GROUP BY (ALL is better), and semantic is not clear (or dangerous) for ORDER BY.

Reply:
oh makes sense